Lilu/Sources/kern_devinfo.cpp

```diff
@@ -206,9 +206,17 @@ void DeviceInfo::grabDevicesFromPciRoot(IORegistryEntry *pciRoot) {
 				continue;
 
 			if (vendor == WIOKit::VendorID::Intel && (code == WIOKit::ClassCode::DisplayController || code == WIOKit::ClassCode::VGAController)) {
-				DBGLOG("dev", "found IGPU device %s", safeString(name));
-				videoBuiltin = obj;
-				requestedExternalSwitchOff |= videoBuiltin->getProperty(RequestedExternalSwitchOffName) != nullptr;
+					DBGLOG("dev", "found IGPU device %s", safeString(name));
+					videoBuiltin = obj;
+					requestedExternalSwitchOff |= videoBuiltin->getProperty(RequestedExternalSwitchOffName) != nullptr;
+			} else if (vendor == WIOKit::VendorID::ATIAMD && (code == WIOKit::ClassCode::DisplayController || code == WIOKit::ClassCode::VGAController)) {
+				uint32_t dev = 0;
+				WIOKit::getOSDataValue(obj, "device-id", dev);
+				dev = (dev & 0xFF00);
+				if ((dev == 0x1600 || dev == 1500) || (dev == 0x9800 || dev == 0x1300) || (dev == 0x9900 || dev == 0x9600)) {
```

Comment:
In general I disagree with hard-coding device IDs in this file. Is it possible to avoid it?